src/main/java/net/neoforged/neoforge/event/level/BlockDropsEvent.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.level;
+
+import com.google.common.base.Preconditions;
+import java.util.List;
+import net.minecraft.core.BlockPos;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.item.ItemEntity;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.enchantment.Enchantments;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.state.BlockState;
+import net.neoforged.bus.api.ICancellableEvent;
+import net.neoforged.neoforge.common.loot.LootModifier;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Fired when a block is broken and the drops have been determined, but before they have been added to the world. This event can be used to manipulate the dropped items and experience.
+ * <p>
+ * No guarantees can be made about the block. It will either have already been removed from the world, or will be removed after the event terminates.
+ * <p>
+ * If you wish to edit the state of the block in-world, use {@link BreakEvent}.
+ */
+public class BlockDropsEvent extends BlockEvent implements ICancellableEvent {
+    @Nullable
+    private final BlockEntity blockEntity;
+    private final List<ItemEntity> drops;
+    @Nullable
+    private final Entity breaker;
+    private final ItemStack tool;
+    private int experience;
+
+    public BlockDropsEvent(ServerLevel level, BlockPos pos, BlockState state, @Nullable BlockEntity blockEntity, List<ItemEntity> drops, @Nullable Entity breaker, ItemStack tool) {
```

Comment:
This should be marked internal.

Reply:
That one was intentionally not marked internal (though I should have placed a javadoc on it)

To simulate block drops without dropping the entities into the world it may be necessary to fire this event outside the context of `CommonHooks#handleBlockDrops`.